jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/ElasticsearchDependenciesJob.java

```diff
@@ -0,0 +1,242 @@
+import io.jaegertracing.spark.dependencies.Utils;
+import io.jaegertracing.spark.dependencies.model.Dependency;
+import io.jaegertracing.spark.dependencies.model.Span;
+import java.net.URI;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.logging.Logger;
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.elasticsearch.spark.rdd.api.java.JavaEsSpark;
+
+/**
+ * @author Pavol Loffay
+ */
+public class ElasticsearchDependenciesJob {
+
+  private static final Logger log = Logger.getLogger(ElasticsearchDependenciesJob.class.getName());
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
+
+    String index = Utils.getEnv("ES_INDEX", "jaeger");
+    String hosts = Utils.getEnv("ES_HOSTS", "127.0.0.1");
+    String username = Utils.getEnv("ES_USERNAME", null);
+    String password = Utils.getEnv("ES_PASSWORD", null);
+
+    final Map<String, String> sparkProperties = new LinkedHashMap<>();
+
+    Builder() {
+      sparkProperties.put("spark.ui.enabled", "false");
+      // don't die if there are no spans
+      sparkProperties.put("es.index.read.missing.as.empty", "true");
+      sparkProperties.put("es.nodes.wan.only", Utils.getEnv("ES_NODES_WAN_ONLY", "false"));
+      sparkProperties.put("es.net.ssl.keystore.location",
+          getSystemPropertyAsFileResource("javax.net.ssl.keyStore"));
+      sparkProperties.put("es.net.ssl.keystore.pass",
+          System.getProperty("javax.net.ssl.keyStorePassword", ""));
+      sparkProperties.put("es.net.ssl.truststore.location",
+          getSystemPropertyAsFileResource("javax.net.ssl.trustStore"));
+      sparkProperties.put("es.net.ssl.truststore.pass",
+          System.getProperty("javax.net.ssl.trustStorePassword", ""));
+    }
+
+    // local[*] master lets us run & test the job locally without setting a Spark cluster
+    String sparkMaster = Utils.getEnv("SPARK_MASTER", "local[*]");
+    // needed when not in local mode
+    String[] jars;
+
+    // By default the job only works on traces whose first timestamp is today
+    long day = Utils.midnightUTC(System.currentTimeMillis());
+
+    /** When set, this indicates which jars to distribute to the cluster. */
+    public Builder jars(String... jars) {
+      this.jars = jars;
+      return this;
+    }
+
+    /** The index prefix to use when generating daily index names. Defaults to "zipkin" */
+    public Builder index(String index) {
+      Utils.checkNoTNull("index", index);
+      this.index = index;
+      return this;
+    }
+
+    public Builder hosts(String hosts) {
+      Utils.checkNoTNull(hosts, "hosts");
+      this.hosts = hosts;
+      sparkProperties.put("es.nodes.wan.only", "true");
```

Comment:
Shouldn't this be on the builder?

Reply:
I am not sure, I took the builder from zipkin. I suppose if it was needed it would be in the builder.